tests/auto/jobs/build.py

```diff
@@ -152,68 +198,101 @@ def clone_pr_repo(job_obj, workdir):
+    logger.info(f'Status of build: {logfile_pass}')
+
+    return logfile_pass
 
 
 def process_logfile(job_obj, ci_log):
+    """
+    Runs after code has been cloned and built
+    Checks to see whether build was successful or failed
+    """
     logger = logging.getLogger('BUILD/PROCESS_LOGFILE')
     fail_string = 'FAIL'
     build_failed = False
     if os.path.exists(ci_log):
-        with open(ci_log) as f:
-            for line in f:
+        with open(ci_log) as fname:
+            for line in fname:
                 if fail_string in line:
                     build_failed = True
-                    job_obj.comment_text_append(f'{line.rstrip(chr(10))}')
+                    job_obj.comment_append(f'{line.rstrip()}')
         if build_failed:
-            job_obj.job_failed(logger, f'{job_obj.preq_dict["action"]}')
-            logger.info('Build failed') 
+            job_obj.send_comment_text()
+            logger.info('Build failed')
         else:
-            logger.info('Build was successful') 
-        return not build_failed 
+            logger.info('Build was successful')
+        return not build_failed
     else:
         logger.critical(f'Could not find {job_obj.machine}'
                         f'.{job_obj.compiler} '
                         f'{job_obj.preq_dict["action"]} log')
         raise FileNotFoundError
+
+
+def process_gen(job_obj, gen_log_loc, gen_log_name):
+    """
+    Runs after a rocoto workflow has been generated
+    Checks to see if an error has occurred
+    """
+    logger = logging.getLogger('BUILD/PROCESS_GEN')
+    gen_log = f'{gen_log_loc}/{gen_log_name}'
+    error_string = 'ERROR'
+    gen_failed = False
+    if os.path.exists(gen_log):
+        with open(gen_log) as fname:
+            for line in fname:
+                if error_string in line:
+                    job_obj.comment_append('Generating Workflow Failed')
+                    gen_failed = True
+                    logger.info('Generating workflow failed')
+                if gen_failed:
+                    job_obj.comment_append(f'{line.rstrip()}')
+
+
+def process_expt(job_obj, expt_dirs):
+    """
+    Runs after a rocoto workflow has been started to run one or more expts
+    Assumes that more expt directories can appear after this job has started
+    Checks for success or failure for each expt
+    """
+    logger = logging.getLogger('BUILD/PROCESS_EXPT')
+    expt_done = 0
+    repeat_count = 72
+    complete_expts = []
+    expt_list = os.listdir(expt_dirs)
+    complete_string = "This cycle is complete"
+    failed_string = "FAILED"
+
+    while (expt_done < len(expt_list)) and repeat_count > 0:
+        time.sleep(300)
+        repeat_count = repeat_count - 1
+        expt_list = os.listdir(expt_dirs)
+        logger.info('Experiment dir after return of end_to_end')
+        logger.info(expt_list)
+        for expt in expt_list:
+            expt_log = expt_dirs + '/' + expt + \
+                '/log/FV3LAM_wflow.log'
```

Comment:
It would be nice to use `os.path.join()` when building paths. It can help with `/` management quite a bit.

Reply:
Changed to that in a couple of places.